File: Java/LevelOrderBottom.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrderBottom(TreeNode root) {
        if(root == null) return new ArrayList<>();
        List<List<Integer>> res = new ArrayList<>();
        Queue<TreeNode> q = new LinkedList<>();
        q.offer(root);

        while(!q.isEmpty()) {
            List<Integer> elem = new ArrayList<>();
            int size = q.size();
            for(int i = 0; i < size; i++) {
                TreeNode n = q.poll();
                elem.add(n.val);
                if(n.left != null) q.offer(n.left);
                if(n.right != null) q.offer(n.right);
            }
            res.add(elem);
        }

        for(int i = 0; i < res.size() / 2; i++) {
            int j = res.size() - 1 - i;
            List<Integer> tmp = res.get(j);
            res.set(j, res.get(i));
            res.set(i, tmp);
        }

        return res;
    }
}

File: Go/level_order_bottom.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func levelOrderBottom(root *TreeNode) [][]int {
    if root == nil {
        return [][]int{}
    }

    var res [][]int
    var q []*TreeNode
    q = append(q, root)

    for len(q) > 0 {
        size := len(q)
        var elem []int
        for i := 0; i < size; i++ {
            n := q[i]
            elem = append(elem, n.Val)
            if n.Left != nil {
                q = append(q, n.Left)
            }
            if n.Right != nil {
                q = append(q, n.Right)
            }
        }
        res = append(res, elem)
        q = q[size:]
    }

    for i, j := 0, len(res) - 1; i < j; i, j = i + 1, j - 1 {
        res[i], res[j] = res[j], res[i]
    }

    return res
}